dev-support/hbase-personality.sh

```diff
@@ -45,6 +45,23 @@ if ! declare -f "yetus_info" >/dev/null; then
 
 fi
 
+# work around yetus overwriting JAVA_HOME from our docker image
+function docker_do_env_adds
```

Comment:
I think this function can be skipped -- instead, pass `--java-home` as an invocation argument to `test-patch`.

Reply:
Can't do that because in the case of the general precommit job (the one that looks at patches on Jira) we don't know what branch we're targeting until Yetus is running.

Without knowing what branch we're on we can't know which jdk to point to.

Yetus will then take care of getting the Dockerfile specific to that branch.

Reply:
Also this is working around what I suspect is a bug in Yetus.

Without this Yetus tells docker to unset any JAVA_HOME defined in the image if the host doesn't have JAVA_HOME set, which doesn't make sense to me.

Reply:
> Without this Yetus tells docker to unset any JAVA_HOME defined in the image if the host doesn't have JAVA_HOME set, which doesn't make sense to me.

This sounds like to a bug to me as well. As far as I can see, the Yetus runtime from the build host should be completely agnostic to the runtime within the docker context -- no need for setting `JAVA_HOME` any more than there's a need for, for example, a `go` executable.

Reply:
tracked via [YETUS-913](https://issues.apache.org/jira/browse/YETUS-913)